habitat/datasets/eqa/mp3d_eqa_dataset.py

```diff
@@ -52,6 +53,19 @@ def __init__(self, config: Config = None) -> None:
         with gzip.open(config.DATA_PATH.format(split=config.SPLIT), "rt") as f:
             self.from_json(f.read())
 
+        self.vocab = VocabFromText(
```

Comment:
The questions and answers should probably have different vocabs.  

The vocab should also be generated once and saved to disk.  This will handle the issue of needing to get the vocab from train for val/test and makes sure vocab is always the same across many different machines.

Reply:
Agreed, `VocabFromText` should be used once to generate the vocab. Than, `VocabDict` class in pythia should be used to load the vocabs easily later on.

Reply:
Ideally, the dataset should download the vocab from a url when it is not present and then you should load it using `VocabDict`